### PR TITLE
[PT FE]: support transformation for case aten::size + aten::__getitem__

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *support aten::__getitem__*

### Tickets:
 - *94276*

### CODE REVIEW

File: src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp [hunk 1]
```diff
@@ -95,6 +95,17 @@ AtenGetItemReplacer::AtenGetItemReplacer() {
             }
             return true;
         }
+        if (auto shape_of = std::dynamic_pointer_cast<opset10::ShapeOf>(input_node)) {
+            // case aten::size as input
+            auto getitem_idx = getitem->input(1).get_source_output();
```
Comment (rkazants) [inline src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp:100]:
```suggestion
            auto getitem_idx = getitem->input_value(1);
```

**Review by rkazants — APPROVED** [review]
@mvafin, please review it.

File: src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp [hunk 2]
```diff
@@ -95,6 +95,17 @@ AtenGetItemReplacer::AtenGetItemReplacer() {
             }
             return true;
         }
+        if (auto shape_of = std::dynamic_pointer_cast<opset10::ShapeOf>(input_node)) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp:98]: I think we can support the case when `__getitem__` is located in body and input is located outside. But there is a problem when `ListConstruct` is located outside. Basically we can translate `__getitem__` to `Gather` if input is `Parameter` and `ListConstruct` (with non zero number of inputs) to `Concat` with axis 0. That should cover majority of cases, but we can still have some issues with that approach.

**Review by mvafin — COMMENTED** [review]
Please add PR description

File: src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp]: moved to translators

File: src/core/src/op/util/multi_subgraph_base.cpp [hunk 3]
```diff
@@ -156,7 +156,9 @@ void ov::op::util::MultiSubGraphOp::validate_and_infer_type_body(
 
         auto body_parameter = params.at(input_description->m_body_parameter_index);
         auto input_partial_shape = input_value(index).get_partial_shape();
+        auto dtype = input_value(index).get_element_type();
```
Comment (mvafin) [inline src/core/src/op/util/multi_subgraph_base.cpp:159]: Could you add test in https://github.com/openvinotoolkit/openvino/blob/master/src/core/tests/type_prop/if.cpp that checks this logic?

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/core/src/op/util/multi_subgraph_base.cpp (hunk 3 above)
Comment (eaidova) [reply src/core/src/op/util/multi_subgraph_base.cpp:159]: added

File: src/core/tests/type_prop/if.cpp [hunk 4]
```diff
@@ -335,3 +335,46 @@ TEST(type_prop, if_scalar_and_1d_static_union) {
...
+    Shape out0_shape{32, 40, 10};
+    auto sh = result0->get_output_shape(0);
+    EXPECT_EQ(sh, out0_shape);
+    // Check that If validation validates both bodies
+    Xt->set_element_type(ov::element::f16);
+    Yt->set_element_type(ov::element::f16);
+    Xe->set_element_type(ov::element::f16);
+    Ye->set_element_type(ov::element::f16);
+    X->set_element_type(ov::element::f16);
+    Y->set_element_type(ov::element::f16);
```
Comment (mvafin) [inline src/core/tests/type_prop/if.cpp:376]:
```suggestion
```
This is not needed here. Only out body types should be provided

File: src/core/tests/type_prop/if.cpp [hunk 5]
```diff
@@ -335,3 +335,46 @@ TEST(type_prop, if_scalar_and_1d_static_union) {
     auto sh = result0->get_output_partial_shape(0);
     EXPECT_EQ(sh, out_shape);
 }
+
+TEST(type_prop, if_element_type_dynamic) {
+    // That which we iterate over
+    auto X = make_shared<op::Parameter>(element::dynamic, Shape{32, 40, 10});
+    auto Y = make_shared<op::Parameter>(element::dynamic, Shape{32, 40, 10});
```
Comment (mvafin) [inline src/core/tests/type_prop/if.cpp:342]:
```suggestion
    auto X = make_shared<op::Parameter>(ov::element::f16, Shape{32, 40, 10});
    auto Y = make_shared<op::Parameter>(ov::element::f16, Shape{32, 40, 10});
```

**Review by mvafin — CHANGES_REQUESTED** [review]

**Review by mvafin — APPROVED** [review]

**Pull request merged**